### ISSUE TITLE
Feature Request - Reformat --help docs

### ISSUE THREAD

**Issue by jerowe** [maros/MooseX-App#43]
Hello,

Thanks for the great library! I don't think there's anything better out there for writing command line apps. ;-)

When using MooseX::Getopt::Long::Usage I get help docs that look like this:

`
➜  ~ mcerunner.pl 
Required option missing: infile
Usage:
    mcerunner.pl [OPTIONS]

Options:
     Required:
         --infile          - Str. File of commands separated by newline. The command 'wait' indicates all previous commands should finish before starting the next one.
     Optional:
         --help -? --usage - Bool. Display the usage message and exit
         --using_mce       - Bool. Default=1. 
         --outdir          - Str. Directory to write out files.
`
I quite like the way they are separated into 'Required' and 'Optional' items, and the formatting of type and defaults.

I've dug around in the code a little, but I don't have a real clear idea of how difficult this would be to implement. I could work on it and generate a PR if that is ok with you.

![screenshot from 2016-06-14 13-03-07](https://cloud.githubusercontent.com/assets/5555526/16037116/28f2405a-3230-11e6-9f9a-473ee9970d40.png)

**Comment by maros** [comment]
Hello,
yes that looks very nice. Splitting up required and optional arguments is probably not a hard thing to do, just have a look at how positional parameters and options are handled in MooseX::App::Meta::Role::Class::Base::command_usage_options and command_usage_command. Maybe you could try this in a separate plugin?
However, unfortunately changing the colouring of individual chunks is hard to accomplish given the current rending mechanism (see MooseX::App::Message). I was planing on rewriting the whole rendering by introducing a simple markup language and having multiple renderers (eg. simple renderer, coloured renderer, ...) consume this markup - but i guess that this will take some time and effort.

**Comment by jerowe** [comment]
@maros, I think I could get some colors without too much trouble by playing with the color plugin. Thanks for pointing me in the direction of code to start. A separate plugin is fine.

**Comment by jerowe** [comment]
I'm going to close this out. Its a nice to have, but not something I think I will ever get around to.

Still love the framework, and I write all my major applications in it. ;-)

**Comment by jerowe** [comment]
I don't know if you appreciate shoutouts, but if you do here are two major applications still under development that use MooseX::App as their commandline framework of choice. 

https://jerowe.gitbooks.io/hpc-runner-command-docs/content/
https://biosails.gitbooks.io/biox-workflow-command-docs/content/
